activesupport/lib/active_support/hash_with_indifferent_access.rb

```diff
@@ -393,7 +393,7 @@ def to_proc
 
     private
       def convert_key(key)
-        key.kind_of?(Symbol) ? key.name : key
+        Symbol == key ? key.name : key
```

Comment:
I think tests are failing due to this typo.

```suggestion
        Symbol === key ? key.name : key
```

But out of curiosity, why prefer `===` over `kind_of?`?  Is there some internal optimization?

Reply:
> why prefer `===` over `kind_of?`? Is there some internal optimization?

Performance wise it's the same, they both end up in `rb_obj_is_kind_of`. That said YJIT has a slightly better codegen for it:

  - https://github.com/ruby/ruby/blob/a0eecfb5bae66470ccf40e27b9193fbf5c76618f/yjit/src/codegen.rs#L4486-L4516
  - https://github.com/ruby/ruby/blob/a0eecfb5bae66470ccf40e27b9193fbf5c76618f/yjit/src/codegen.rs#L4368-L4418

But it's not the predominent reason. It's more to not conflict with various proxy objects and such.

Reply:
> It's more to not conflict with various proxy objects and such.

I'm not familiar with that kind of issue.  Would you mind giving me an example?

(I ask because I usually write `is_a?`, but if there are reasons to prefer `===`, I would like to know.)

Reply:
> I'm not familiar with that kind of issue. Would you mind giving me an example?

Sure, but keep in mind that this is only relevant for code exposed to the wild, absolutely not relevant to application code.

When writing proxy or delegator objects, you often inherit from `BasicObject`, which doesn't implement `is_a?`:

```
>> BasicObject.new.is_a?(Symbol)
(irb):1:in `<main>': undefined method `is_a?' for an instance of BasicObject (NoMethodError)
```

So you might just get a `NoMethodError`, if the implemented didn't think about it. Now the reasons of putting such object as a key in a Hash are rather limited, and may never have happened ever as AFAIK there is no bug report for it, but it's a bit of an habit I developed when writing framework/library code.

Reply:
I see.  Thank you! :heart: